api/app.py

```diff
@@ -15,12 +18,138 @@ def get_users():
            200)
     return resp
 
+
 @app.route('/api/user/<int:user_id>')
 def get_user(user_id):
 
     user = User.query.get(user_id)
     resp = make_response(jsonify(user.to_dict_for_json), 200)
     return resp
 
+
+@app.route('/api/opportunities')
+def get_opportunities():
+    """ get offers and requests """
+    
+    requests = App_Request.query.all()
+    offers = Offer.query.all()
+
+    resp = make_response(jsonify(
+                        {'requests': [app_request.to_dict_for_json for app_request in requests],
+                         'offers': [offer.to_dict_for_json for offer in offers]
+                        }))
+
+    return resp
+
+
+@app.route('/api/requests')
+def get_requests():
+    
+    requests = App_Request.query.all()
+    resp = make_response(jsonify([app_request.to_dict_for_json for app_request in requests]), 
+           200)
+    return resp
+
+
+@app.route('/api/request/<int:request_id>')
+def get_request(request_id):
+
+    app_request = App_Request.query.get(request_id)
+    resp = make_response(jsonify(app_request.to_dict_for_json), 200)
+    return resp
+
+
+@app.route('/api/request', methods=['POST'])
+def create_update_request():
+
+    content = request.get_json()
+    request_id = content['request_id']
+    requester_user_id = content['requester_user_id']
+    volunteer_user_id = content['volunteer_user_id']
```

Comment:
I think we won't have this when we first post the request

Reply:
Yeah, I agree with the volunteer_user_id. We should have the requester_user_id though, right (whoever is creating the request).

Reply:
Yup! So basically, I would just have the UI not send it and y'all can set to null on your end.